File: backend/app/schemas/zone.py
from __future__ import annotations
from pydantic import BaseModel


class ZoneSummary(BaseModel):
    id: str
    name: str
    region: str
    total_equipment: int
    total_customers: int
    critical_count: int
    high_count: int
    medium_count: int
    low_count: int
    zone_risk_score: float
    zone_risk_level: str
    outage_probability: float | None = None


class ZoneDetail(ZoneSummary):
    equipment_ids: list[str] = []

File: frontend/src/types/zone.ts
// zone.ts
export interface ZoneSummary {
  id: string;
  name: string;
  region: string;
  total_equipment: number;
  total_customers: number;
  critical_count: number;
  high_count: number;
  medium_count: number;
  low_count: number;
  zone_risk_score: number;
  zone_risk_level: string;
  outage_probability?: number;
}
